src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,227 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+
+import { deleteTodo, patchTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { AppContext } from '../AppContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo }) => {
+  const {
+    id,
+    title,
+    completed,
+  } = todo;
+
+  const [isEditingMode, setIsEditingMode] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const input = useRef<HTMLInputElement>(null);
+
+  const {
+    allTodos,
+    setAllTodos,
+    showError,
+    setShouldShowError,
+    loadingTodosIds,
+    setLoadingTodosIds,
+  } = useContext(AppContext);
+
+  const isLoadingState = useMemo(() => (
+    loadingTodosIds.includes(id)
+  ), [loadingTodosIds]);
+
+  const deleteTodoFromServer = useCallback(async () => {
+    try {
+      setShouldShowError(false);
+      setLoadingTodosIds(prevIds => [...prevIds, id]);
+
+      await deleteTodo(id);
+
+      setAllTodos(prevTodos => prevTodos.filter(prevTodo => (
+        prevTodo.id !== id
+      )));
+    } catch {
+      showError('Unable to delete a todo');
+    } finally {
+      setLoadingTodosIds([0]);
+    }
+  }, [loadingTodosIds, allTodos]);
+
+  const toggleTodoStatus = useCallback(async () => {
+    try {
+      setShouldShowError(false);
+      setLoadingTodosIds(prevIds => [...prevIds, id]);
+
+      await patchTodo(id, { completed: !completed });
+
+      setAllTodos(prevTodos => prevTodos.map(prevTodo => {
+        return (prevTodo.id === id)
+          ? {
+            ...prevTodo,
+            completed: !completed,
+          }
+          : prevTodo;
+      }));
```

Comment:
Create method for update todo and reuse it for both updating status and title, this logic is same